Raymond Mule - Signoff Request - Wireless Communication Subsystem

Raymond-Mule opened the pull request:
Raymond Mule - Signoff Request - Wireless Communication Subsystem

Raymond-Mule committed:
Raymond Mule - Signoff Request - Wireless Communication Subsystem

Michaelwwest98 committed:
Delete Heartbeat-Respiratory-Subsystem.md

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,37 @@
+## Constraints- 
+| Constraint | Source |
+|------------|--------|
+| 1. The system shall be able to communicate with the computing subsystem from a minimum distance of 700-meters| Analyzed from DARPA's constraints |
```
Give more detail below the constraints section here about how you derived this. Don't give the explanation in the analysis section.

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,37 @@
+| Constraint | Source |
+|------------|--------|
+| 1. The system shall be able to communicate with the computing subsystem from a minimum distance of 700-meters| Analyzed from DARPA's constraints |
+| 2. The system shall send the results from the computing subsystem to user interface subsystem in close to real time | From DARPA's constraints [1] |
```
Ambiguous

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,37 @@
+|------------|--------|
+| 1. The system shall be able to communicate with the computing subsystem from a minimum distance of 700-meters| Analyzed from DARPA's constraints |
+| 2. The system shall send the results from the computing subsystem to user interface subsystem in close to real time | From DARPA's constraints [1] |
+| 2. The system shall send results from the computing subsystem to user interface subsystem at a maximum time of two seconds | Analyzed from DARPA's constraints |
```
Conflicts with other constraints

JesseTNRoberts reviewed (commented):
Do not talk about darpa's constraints in the analysis section. Show that the stated constraints are met by your design.

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Updated constraints and analysis sections.

JesseTNRoberts reviewed (commented):
The 700 meters number does not make sense.

You are not allowed to represent a single device more than once. You will have to work with who ever drew the Jetson first and combine the schematics in some manner.

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Apologies about the schematic, I have read your comment concerning it, but I have been unable to meet with my team member that has been working on the computing system this week due to our finals schedules.

JesseTNRoberts reviewed (commented):
You need to look into transmission power that is allowed. Standards and regulatory bodies.

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Added a new schematic with slightly different components, and I also added a standard covering the maximum power output of the device.

JesseTNRoberts reviewed (commented):
how much data and how much bandwidth is required?

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Addressed comments, but I am not completely sure what you meant by data. I would like to talk to you about this comment on Monday during lecture hours to get some clarification,

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -10,6 +10,8 @@ The function of the wireless communication subsystem is to wirelessly transport
 | 3. The entire system shall weigh less than six pounds| From conceptual design |
 | 4. Unlicensed devices that operate in the 915 MHz band allow for a maximum power output of up to 30 dBm or 1 Watt | From Federal Communications Commission (FCC) |
+| 5. The bandwidth of the device must be 125 kHz or 500 kHz | From further analysis of the subsystem |
+| 6. The system must be able to transmit at least 30 bits of data | From further analysis of the subsystem |
```
per second? per packet?

JesseTNRoberts reviewed (commented):
This seems difficult to accomplish. Does it really have the long of a range at the needed bandwidth?

JesseTNRoberts reviewed (commented):
How are you plugging in multiple USBs to a single port?

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Addressed the comments concerning data by fixing constraint #6. The data rate is specified in the analysis section #2. The schematic that makes it seem that two devices are being plugged into the Jetson Nano via USB port is actually only one device. On the schematic, there is a parenthesis on the component RAKDAP1 that specifies if you are looking at the top of the device or the bottom of the device. I did this since there are inputs on the bottom of the RAKDAP1 and the top of it as well.

JesseTNRoberts reviewed (commented):
It has to be compatible with the Jetson nano. I don't see anything that addresses that. Are we sure that it will work with the cpu? How do we know?

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Addressed latest comment.

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -58,6 +59,8 @@ The RAK811 transceiver has a very high sensitivity of -148 dBm, enabling it to t
 
 6. The RAK811 Breakout Board will be sending data to and from the computing subsystem to the user interface subsystem. The size of data that will be sent from the computing subsystem to the user interface subsystem and back will approximately be around 30 bits. The maximum payload size that the RAK811 transceiver can send in one data packet is 52 bytes which is signifacntly greater than the neede ...
 
+7. The RAK811 Breakout Board must be connected to the computing subsystem in order to transmit data from the computing subsystem to the user interface subsystem. In order connect the RAK811 Breakout Board and the computing subsystem, a RAKDAP1 is used to connect the transceiver to the computing subsystem. The RAKDAP1 is ewuipped with a USB that enables it to connect to any device that possess a U ...
```
This doesn't really address compatibility.

JesseTNRoberts replied:
Prove that it will work with what it needs to work with the computer you intend to plug it to.

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Rewrote point 7 of the analysis section to address linux compatibility of the subsystem.

JesseTNRoberts reviewed (commented):
A debug tool is not how this device will interact once it is programmed. A debug tool allows you to debug.

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Addressed previous comment clarifying that the transceiver is compatible with Linux.

JesseTNRoberts reviewed (commented):
The design is confusing. Why are there 4 of the same device shown? (2 I understand) 

Also, I don't want paragraphs of content in the analysis. Say exactly what must be said to show that the constraint is met and that the system will work and no more.

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Added an explanation regarding the need for two UART to USB converters. Also trimmed down analysis.

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Updated schematic

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,71 @@
+## Analysis-
+1.  The RAK811 transceiver has a range of 15 kilometers, which fulfills the constraint of a 9 kilometer distance.
+
+2. The data that will be sent from the computing system to the user interface will be composed of measurements obtained from the sensing subsystem as well as a simple graphic that displays the START status of a person. This data will most likely hold a data size of at least several kilobytes. The RAK811 has an average data rate of 300 kbps. Using this radio frequency module with an average data r ...
```
This is not a tight enough estimation on how much data will be sent.

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,71 @@
+
+2. The data that will be sent from the computing system to the user interface will be composed of measurements obtained from the sensing subsystem as well as a simple graphic that displays the START status of a person. This data will most likely hold a data size of at least several kilobytes. The RAK811 has an average data rate of 300 kbps. Using this radio frequency module with an average data r ...
+
+3.  The weight of the RAK811 LPWAN breakout module the RAKDAP1 debug and flash tool that will be attached to the sensing system weighs approximately 24.4 grams or 0.0538 pounds which is a small fraction of the maximum six pounds the entire system must weigh.
```
I don't understand the logic here. many small fractions add up to a large quantity.

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,71 @@
+
+3.  The weight of the RAK811 LPWAN breakout module the RAKDAP1 debug and flash tool that will be attached to the sensing system weighs approximately 24.4 grams or 0.0538 pounds which is a small fraction of the maximum six pounds the entire system must weigh.
+
+4. The RAK811 Breakout Board operates in the 915 Mhz band and the maximum power output that the RAK811 Breakout Board is 100mW or 20 dBm, which falls below the safety standard set by the FCC concerning maximum power output of an unlicensed device operating in the 915 MHz frequency band.
```
What safety standard?

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,71 @@
+
+4. The RAK811 Breakout Board operates in the 915 Mhz band and the maximum power output that the RAK811 Breakout Board is 100mW or 20 dBm, which falls below the safety standard set by the FCC concerning maximum power output of an unlicensed device operating in the 915 MHz frequency band.
+
+5. The RAK811 Breakout board as previously stated, must operate in the 915 MHz band which is the frequency band for United States. This means the bandwidth that is required for the device is either 125 kHz or 500 kHz. On the RAK811 data sheet, it does not specify a specific bandwidth that the device possesses. All that is stated on the data sheet is that the RAK811 Breakout Board supports US915 o ...
```
the frequency band for the US? what does that mean? 

If it doesn't state a bandwidth how do you know it will be sufficient?

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,71 @@
+
+5. The RAK811 Breakout board as previously stated, must operate in the 915 MHz band which is the frequency band for United States. This means the bandwidth that is required for the device is either 125 kHz or 500 kHz. On the RAK811 data sheet, it does not specify a specific bandwidth that the device possesses. All that is stated on the data sheet is that the RAK811 Breakout Board supports US915 o ...
+
+6. The size of data that will be sent from the computing subsystem to the user interface subsystem and back will approximately be around 30 bits. The maximum payload size that the RAK811 transceiver can send in one data packet is 52 bytes which is signifacntly greater than the needed amount. 
```
Seems to be a contradiction.

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,71 @@
+
+6. The size of data that will be sent from the computing subsystem to the user interface subsystem and back will approximately be around 30 bits. The maximum payload size that the RAK811 transceiver can send in one data packet is 52 bytes which is signifacntly greater than the needed amount. 
+
+7. Within the RAK811 Quick Start Guide documentation, there is a list specifying the required hardware componetes needed for the transceiver to function properly. One of these components specified in the list is a Linux computer which means that the RAK811 transceiver is compatible with Linux based systems. 
```
This isn't sufficient. You need to research how to find out if software is compatible.

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Updated weight constraint, and updated analysis section.

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,73 @@
+## Analysis-
+1.  The RAK811 transceiver has a range of 15 kilometers, which fulfills the constraint of a 9 kilometer distance.
+
+2. The data that will be sent from the computing system to the user interface will be approximately 30 bits in size according to the designer of the computing subsystem. The RAK811 has an average data rate of 300 kbps meaning that the data from the computing subsystem will be able to reach the user interface subsystem within the two second minimum time frame specified in the constraints.
```
How do we know? How long will it take at maximum between receiving two packets of data?

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,73 @@
+
+4. The RAK811 Breakout Board operates in the 915 Mhz band and the maximum power output that the RAK811 Breakout Board is 100mW or 20 dBm, which falls below the safety standard of 30 dBm or 1 watt set by the FCC Part 15 Subpart C.
+
+5. The RAK811 Breakout board as previously stated, must operate in the 915 MHz band which is the frequency band for United States. This means the bandwidth that is required for the device is either 125 kHz or 500 kHz. On the RAK811 data sheet, it states that the bandwidth of the device is able to be configured. The minimum bandwidth of the transceiver is 125 kHz and the maximum is 500 kHz. This f ...
```
Again, what is the frequency band for the US? That statement does not make sense. Further, why is there a bandwidth constraint?

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -0,0 +1,73 @@
+
+7. Within the RAK811 Quick Start Guide documentation, there is a list specifying the required hardware components needed for the transceiver to function properly. One of these components specified in the list is a Linux computer which means that the RAK811 transceiver is compatible with Linux based systems.
+  
+From there, the manufacturer provides a serial port tool allowing the user to send AT commands to the RAK811 transceiver and view replies from the console   output. This software given by the manufacturer is provided as an Ubuntu file which is compatible with Linux based systems.
```
Ubuntu is a generic name for an operating system. Which version does it require? How do you know that will work with the computing subsystem?

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Changed a constraint referring to unlicensed frequency bands. Addressed analysis concerning data rate, unlicensed frequency band, and linux software compatibility.

Raymond-Mule committed:
Update Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md

Made some quick edits on contraints and analysis concerning unlicensed frequency bands

JesseTNRoberts commented on Documentation/Signoffs/Raymond Mule-Signoff-Wireless Communication Subsystem(Updated).md:
```diff
@@ -49,19 +49,23 @@ The RAK811 transceiver has a very high sensitivity of -148 dBm, enabling it to t
+2. The data that will be sent from the computing system to the user interface will be approximately 30 bits in size according to the designer of the computing subsystem. The RAK811 has an average data rate of 300 kbps.
 
-3.  The weight of the RAK811 LPWAN breakout module the RAKDAP1 debug and flash tool that will be attached to the sensing system weighs approximately 24.4 grams or 0.0538 pounds which is a signaficantly less than the maximum 1.2 pounds the entire subsystem must weigh.
+The amount of time that it will take for 2 data packages to be sent to the computing system to the user interface and back, or sending 30 bits of data back and forth will take 0.0002 seconds with the data rate of 300 kbps that the transceiver possesses.
```
How did you arrive at this number? Show the work to get there.
